### PR TITLE
sink (ticdc): add a non-blocking API to sink

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: ref #4222 

### What is changed and how it works?
Add a API: `TryEmitRowChangedEvents(ctx context.Context, rows ...*model.RowChangedEvent) (bool, error)` to `Sink`.

### Check List 

Tests 

Code changes

 - Has exported function/method change
 - Has interface methods change

Side effects

 - Increased code complexity

Related changes

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sink/sink.go [hunk 1]
```diff
@@ -39,6 +39,8 @@ type Sink interface {
 	// FIXME: some sink implementation, they should be.
```
Comment (overvenus) [inline cdc/sink/sink.go:39]:
```suggestion
	// FIXME: some sink implementation is not thread-safe, but they should be.
```

Please correct similar comments in this file, thanks!

File: cdc/sink/sink.go [hunk 2]
```diff
@@ -39,6 +39,8 @@ type Sink interface {
...
 	EmitRowChangedEvents(ctx context.Context, rows ...*model.RowChangedEvent) error
 
+	// TryEmitRowChangedEvents is thread-safe and non-blocking.
```
Comment (overvenus) [inline cdc/sink/sink.go:42]: Please comment thread-safety of the method.

**Review by overvenus — COMMENTED** [review]
rest LGMT

**Review by overvenus — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Pull request merged**